### ISSUE TITLE
Velero backup - create in cloud-platform-components tf

### ISSUE THREAD

**Issue by sablumiah** [ministryofjustice/cloud-platform#1355]

**Comment by sablumiah** [comment]
Create bucket and role:
https://velero.io/docs/v1.0.0/aws-config/
Helm Chart:
https://github.com/helm/charts/tree/master/stable/velero

Github:
https://github.com/vmware-tanzu/velero

Good Articles to read:
https://blog.kubernauts.io/backup-and-restore-of-kubernetes-applications-using-heptios-velero-with-restic-and-rook-ceph-as-2e8df15b1487

Helm chart
```
##
## Configuration settings that directly affect the Velero deployment YAML.
##

# Details of the container image to use in the Velero deployment & daemonset (if
# enabling restic). Required.
image:
  repository: gcr.io/heptio-images/velero
  tag: v1.1.0
  pullPolicy: IfNotPresent

# Annotations to add to the Velero deployment's pod template. Optional.
#
# If using kube2iam or kiam, use the following annotation with your AWS_ACCOUNT_ID
# and VELERO_ROLE_NAME filled in:
#  iam.amazonaws.com/role: arn:aws:iam::<AWS_ACCOUNT_ID>:role/<VELERO_ROLE_NAME>
podAnnotations: {}

# Resource requests/limits to specify for the Velero deployment. Optional.
resources: {}

# Init containers to add to the Velero deployment's pod spec. Optional.
initContainers: []
  # - name:
  #   image:
  #   volumeMounts:
  #     - name: plugins
  #       mountPath: /target

# Tolerations to use for the Velero deployment. Optional.
tolerations: []

# Node selector to use for the Velero deployment. Optional.
nodeSelector: {}

# Settings for Velero's prometheus metrics. Disabled by default.
metrics:
  enabled: true
  scrapeInterval: 30s

  # Pod annotations for Prometheus
  podAnnotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "8085"
    prometheus.io/path: "/metrics"

  serviceMonitor:
    enabled: true
    additionalLabels: {}

##
## End of deployment-related settings.
##


##
## Parameters for the `default` BackupStorageLocation and VolumeSnapshotLocation,
## and additional server settings.
##
configuration:
  # Cloud provider being used (e.g. aws, azure, gcp).
  provider: aws

  # Parameters for the `default` BackupStorageLocation. See
  # https://velero.io/docs/v1.0.0/api-types/backupstoragelocation/
  backupStorageLocation:
    # Cloud provider where backups should be stored. Usually should
    # match `configuration.provider`. Required.
    name: aws
    # Bucket to store backups in. Required.
    bucket: cloud-platform-velero-bucket-test
    # Prefix within bucket under which to store backups. Optional.
    prefix: 
    # Additional provider-specific configuration. See link above
    # for details of required/optional fields for your provider.
    config: 
      region: eu-west-2
    #  s3ForcePathStyle:
    #  s3Url:
    #  kmsKeyId:
    #  resourceGroup:
    #  storageAccount:
    #  publicUrl:

  # Parameters for the `default` VolumeSnapshotLocation. See
  # https://velero.io/docs/v1.0.0/api-types/volumesnapshotlocation/
  volumeSnapshotLocation:
    # Cloud provider where volume snapshots are being taken. Usually
    # should match `configuration.provider`. Required.,
    name: aws
    # Additional provider-specific configuration. See link above
    # for details of required/optional fields for your provider.
    config:
      region: eu-west-2
  #    apitimeout:
  #    resourceGroup:
  #    snapshotLocation:
  #    project:

  # These are server-level settings passed as CLI flags to the `velero server` command. Velero
  # uses default values if they're not passed in, so they only need to be explicitly specified
  # here if using a non-default value. The `velero server` default values are shown in the
  # comments below.
  # --------------------
  # `velero server` default: 1m
  backupSyncPeriod:
  # `velero server` default: 1h
  resticTimeout:
  # `velero server` default: namespaces,persistentvolumes,persistentvolumeclaims,secrets,configmaps,serviceaccounts,limitranges,pods
  restoreResourcePriorities:
  # `velero server` default: false
  restoreOnlyMode:

  # additional key/value pairs to be used as environment variables such as "AWS_CLUSTER_NAME: 'yourcluster.domain.tld'"
  extraEnvVars: {}

##
## End of backup/snapshot location settings.
##


##
## Settings for additional Velero resources.
##

# Whether to create the Velero cluster role binding.
rbac:
  create: true

# Information about the Kubernetes service account Velero uses.
serviceAccount:
  server:
    create: true
    name:

# Info about the secret to be used by the Velero deployment, which
# should contain credentials for the cloud provider IAM account you've
# set up for Velero.
credentials:
  # Whether a secret should be used as the source of IAM account
  # credentials. Set to false if, for example, using kube2iam or
  # kiam to provide IAM credentials for the Velero pod.
  useSecret: true
  # Name of a pre-existing secret (if any) in the Velero namespace
  # that should be used to get IAM account credentials. Optional.
  existingSecret: 
  # Data to be stored in the Velero secret, if `useSecret` is
  # true and `existingSecret` is empty. This should be the contents
  # of your IAM credentials file.
  secretContents:
    cloud: |
      [default]
      aws_access_key_id=
      aws_secret_access_key=

# Wheter to create volumesnapshotlocation crd, if false => disable snapshot feature
snapshotsEnabled: true

# Whether to deploy the restic daemonset.
deployRestic: false

restic:
  podVolumePath: /var/lib/kubelet/pods
  privileged: false
  # Resource requests/limits to specify for the Restic daemonset deployment. Optional.
  resources: {}

# Backup schedules to create.
# Eg:
# schedules:
#   mybackup:
#     schedule: "0 0 * * *"
#     template:
#       ttl: "240h"
#       includedNamespaces:
#        - foo
schedules:
  allnamespacebackup:
    schedule: "0/30 * * * *"

# Velero ConfigMaps.
# Eg:
# configMaps:
#   restic-restore-action-config:
#     labels:
#       velero.io/plugin-config: ""
#       velero.io/restic: RestoreItemAction
#     data:
#       image: gcr.io/heptio-images/velero-restic-restore-help
configMaps: {}

##
## End of additional Velero resource settings.
##
```

**Comment by pwyborn** [comment]
Cluster:
![Screen Shot 2019-11-20 at 13.39.29.png](https://images.zenhubusercontent.com/5cd14e317c13c23a96974cd5/c85f3d3c-53b5-4e47-966d-551367e5cbf8)

Workspace:
![Screen Shot 2019-11-20 at 13.45.36.png](https://images.zenhubusercontent.com/5cd14e317c13c23a96974cd5/daee34f5-2a7e-4322-ba38-21f02c4256e9)

origin branch:
![Screen Shot 2019-11-20 at 13.47.12.png](https://images.zenhubusercontent.com/5cd14e317c13c23a96974cd5/6e33aee7-15e4-4595-a276-b9dd1bafccb1)

Command:
![Screen Shot 2019-11-20 at 13.49.01.png](https://images.zenhubusercontent.com/5cd14e317c13c23a96974cd5/ed287474-62b8-47a5-ade7-bdfb8ef4a423)

Namespace: "velero" created terraform
S3 bucket: "cloud-platform-velero-bucket-test" created terraform
Role and inline policy "velero.es-test-pk.cloud-platform.service.justice.gov.uk" created terraform/kiam 
Error Message:
![Screen Shot 2019-11-20 at 13.50.48.png](https://images.zenhubusercontent.com/5cd14e317c13c23a96974cd5/6099c736-b935-468a-ac4b-66a21cd80a11)

**Comment by pwyborn** [comment]
Tuesday 26th November 16:35. Hi Sablu - this branch is where I have got to with Velero:
https://github.com/ministryofjustice/cloud-platform-infrastructure/tree/velero-cloud-platform-components
Velero installed in the branch “es-test-pk”. I tested it by deleting and restoring the monitoring namespace. All seemed to work. I am off until Friday now (don’t know if you will have time to have a look at it, and take it further?)
Managed to expose my key in the process (forgot to delete it from earlier!) set off alarm bells/emails etc!
